dlp/inspectWithCustomRegex.js

```diff
@@ -0,0 +1,145 @@
+
+function main(
+  projectId,
+  string,
+  minLikelihood,
+  maxFindings,
+  infoTypes,
+  customInfoTypes,
+  includeQuote
+) {
+  [infoTypes, customInfoTypes] = transformCLI(infoTypes, customInfoTypes);
+  // [START dlp_inspect_custom_regex]
+  // Imports the Google Cloud Data Loss Prevention library
+  const DLP = require('@google-cloud/dlp');
+
+  // Instantiates a client
+  const dlp = new DLP.DlpServiceClient();
+
+  // The project ID to run the API call under
+  // const projectId = 'my-project';
+
+  // The string to inspect
+  // const string = 'Patients MRN 444-5-22222';
+
+  // The minimum likelihood required before returning a match
+  // const minLikelihood = DLP.protos.google.privacy.dlp.v2.Likelihood.POSSIBLE;
+
+  // The maximum number of findings to report per request (0 = server maximum)
+  // const maxFindings = 0;
+
+  // The infoTypes of information to match
+  // See https://cloud.google.com/dlp/docs/concepts-infotypes for more information
+  // about supported infoTypes.
+  // const infoTypes = [{ name: 'EMAIL_ADDRESS' }];
+
+  // The customInfoTypes of information to match
+  // const customInfoTypes = [{ infoType: { name: 'DICT_TYPE' }, dictionary: { wordList: { words: ['foo', 'bar', 'baz']}}},
+  //   { infoType: { name: 'REGEX_TYPE' }, regex: {pattern: '\\(\\d{3}\\) \\d{3}-\\d{4}'}}];
+
+  // Whether to include the matching string
+  // const includeQuote = true;
+
+  async function inspectWithCustomRegex() {
+    // Construct item to inspect
+    const item = {
+      byteItem: {
+        type: DLP.protos.google.privacy.dlp.v2.ByteContentItem.BytesType
+          .TEXT_UTF8,
+        data: Buffer.from(string, 'utf-8'),
+      },
+    };
+
+    // Assigns likelihood to each match
+    customInfoTypes = customInfoTypes.map(customInfoType => {
+      customInfoType.likelihood =
+        DLP.protos.google.privacy.dlp.v2.Likelihood.POSSIBLE;
+      return customInfoType;
+    });
+
+    // Construct request
+    const request = {
+      parent: `projects/${projectId}/locations/global`,
+      inspectConfig: {
+        infoTypes: infoTypes,
+        customInfoTypes: customInfoTypes,
+        minLikelihood: minLikelihood,
+        includeQuote: includeQuote,
+        limits: {
+          maxFindingsPerRequest: maxFindings,
+        },
+      },
+      item: item,
+    };
+
+    // Run request
+    const [response] = await dlp.inspectContent(request);
+    const findings = response.result.findings;
+    if (findings.length > 0) {
+      console.log('Findings:');
+      findings.forEach(finding => {
+        if (includeQuote) {
+          console.log(`\tQuote: ${finding.quote}`);
+        }
+        console.log(`\tInfo type: ${finding.infoType.name}`);
+        console.log(`\tLikelihood: ${finding.likelihood}`);
+      });
+    } else {
+      console.log('No findings.');
+    }
+  }
+  inspectWithCustomRegex();
+  // [END dlp_inspect_custom_regex]
+}
+
+main(...process.argv.slice(2));
```

Comment:
This should be after process.on to avoid missing synchronous promise rejections.

Reply:
Is there any use case where this can happen? I have updated the code as you mentioned but during testing, I found the same results.

Reply:
I don't think it happens right now, but I could imagine in the future there might be synchronous validation of requests (e.g. bad enum values might throw before even making a network request).